### PR TITLE
Support postsubmit checkruns for plugins/packages

### PR DESCRIPTION
**Pull request by keyonghan**
Fixes: https://github.com/flutter/flutter/issues/106902

This PR enables checkrun support for plugins and packages:
1) create checkrun when commit lands
2) update checkrun when builds finish

This PR has been validated with cocoon repo:
Screen Shot 2022-10-21 at 2 44 20 PM

/cc @stuartmorgan

### CODE REVIEW

File: app_dart/test/request_handlers/postsubmit_luci_subscription_test.dart [hunk 1]
```diff
@@ -145,4 +150,24 @@ void main() {
...
+      parent: commit,
+    );
+    config.db.values[task.key] = task;
+
+    tester.message = createBuildbucketPushMessage(
+      'COMPLETED',
+      result: 'SUCCESS',
+      builderName: 'Linux Packages',
+      userData:
+          '{\\"task_key\\":\\"${task.key.id}\\", \\"commit_key\\":\\"${task.key.parent?.id}\\", \\"repo_owner\\": \\"flutter\\", \\"repo_name\\": \\"packages\\"}',
```
Comment (ricardoamador) [inline app_dart/test/request_handlers/postsubmit_luci_subscription_test.dart:168]: Could you use ''' string ''' instead of all the escaping here?

**Review by ricardoamador — APPROVED** [review]
LGTM

File: app_dart/test/request_handlers/postsubmit_luci_subscription_test.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/test/request_handlers/postsubmit_luci_subscription_test.dart:168]: This is to mock the json decoded data:
```
{\"task_key\":\"4507531199512576\", \"commit_key\":\"flutter/flutter/master/87f88734747805589f2131753620d61b22922822\", \"repo_owner\": \"flutter\", \"repo_name\": \"packages\"}
```

Comment (ricardoamador) [reply app_dart/test/request_handlers/postsubmit_luci_subscription_test.dart:168]: Ah gotcha. Might be good to add a note if it is expected in that format with escapes.

Comment (keyonghan) [reply app_dart/test/request_handlers/postsubmit_luci_subscription_test.dart:169]: Done.

**Comment by stuartmorgan** [comment]
Awesome, thanks! I see this in the last flutter/plugin commit :)

**Pull request merged**